libzeth/circuits/circuit_constants.hpp

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) 2015-2020 Clearmatics Technologies Ltd
```

Comment:
These are "protocol constants" rather than "circuit constants". We also use them in the client etc (but since the client is in Python, we have a separate file). I would move these outside of the `circuits` folder and rename the file accordingly

Reply:
Yes, broader than the circuit. And related to the types in `circuit_types.hpp`, which could also be considered similar constants or configuration of the protocol. I've renamed it to `zeth_constants.hpp` and moved it to `libzeth`, which doesn't seem like the right place (since it's not relevant ot `core`, `snarks` or `serialization`), but it seems like overkill to create a `config` or `constants` directory for just this file, and if we wannted to put `circuit_typees.hpp` in there, `config` would have to depend on `circuits`, creating a circuilar dependency.

One option would be to have a `gadgets` dir with just the gadgets for the various commitments, annd then a `zeth` dir (or some other name, broader than circuits) which could contains these constants, the choices for Hash, HashTree, etc and the final `jonisplit` gadget.

For now, I'll keep it in the `libzeth`, but let me know if you have a better ideaa or a strong opinion about it.

Reply:
> For now, I'll keep it in the libzeth, but let me know if you have a better ideaa or a strong opinion about it.

Yes I'd just move it up the file hierarchy and rename accordingly for now